zerver/webhooks/papertrail/view.py

```diff
@@ -16,10 +17,18 @@
 ```
 """.strip()
 
+
+def ensure_keys(name: str, data: Any) -> None:
+    if 'events' in data and 'saved_search' in data:
+        saved_search = data['saved_search']
+        if 'name' in saved_search and 'html_search_url' in saved_search:
+            return
+    raise InvalidJSONError("Malformed JSON")
```

Comment:
We only use `InvalidJSONError` to indicate a problem with JSON decoding itself. If we’ve gotten here, the JSON isn’t malformed. Use `JsonableError(_("Missing expected keys"))` or something.

Reply:
Yeah, I wasn't sure about using the `InvalidJSONError` here, but I used it because it triggers a message to the bot owner with text that looks like `Hi there! It looks like you tried to setup the Zulip {webhook_name} integration, but didn't correctly configure the webhook to send data in the JSON format that this integration expects!`. But, I guess it's not really something that the bot owner did wrong if keys are missing. 

I'll fix this!

Reply:
The critical decision to make when choosing which exception to use is who should fix a problem -- we should only ping the user if it's their fault.    (Which I suppose missing keys might be, if the issue is they are doing HTTP posts from some service that isn't papertrail).